.evergreen/config.yml

```diff
@@ -1,48 +1,191 @@
+# Mark a failure as a system/bootstrap failure (purple box) rather then a task
+# failure by default.
+# Actual testing tasks are marked with `type: test`
+command_type: system
+
+# Protect ourselves against rogue test case, or curl gone wild, that runs forever
+exec_timeout_secs: 3600
+
+# When a task that used to pass starts to fail, go through all versions that may have been skipped
+# to detect when the task started failing
+stepback: true
+
+#######################################
+#              Functions              #
+#######################################
 functions:
-  fetch source:
-    # Executes git clone
+  "fetch source":
+    # Executes git clone and applies the submitted patch, if any.
     - command: git.get_project
       params:
         directory: src
-    # Applies the submitted patch, if any
-    - command: git.apply_patch
 
-  exec script:
-    - command: shell.exec
-      type: test
+  "create expansions":
+    - command: subprocess.exec
+      params:
+        working_dir: src
+        binary: bash
+        args:
+          - .evergreen/create-expansions.sh
+        include_expansions_in_env:
+          - is_patch
+          - version_id
+          - project
+    - command: expansions.update
+      params:
+        file: src/expansion.yml
+
+  "prepare resources":
+    - command: subprocess.exec
       params:
         working_dir: src
-        shell: bash
-        add_to_path:
-          - .evergreen
+        binary: bash
+        args:
+          - .evergreen/fetch-drivers-tools.sh
+        include_expansions_in_env:
+          - DRIVERS_TOOLS
+    - command: shell.exec
+      params:
+        script: |
+          ${PREPARE_SHELL}
+          echo "{ \"releases\": { \"default\": \"$MONGODB_BINARIES\" }}" > $MONGO_ORCHESTRATION_HOME/orchestration.config
+
+  "bootstrap mongo-orchestration":
+    - command: subprocess.exec
+      params:
+        binary: sh
+        args:
+          - ${DRIVERS_TOOLS}/.evergreen/run-orchestration.sh
+        include_expansions_in_env:
+          - AUTH
+          - SSL
+          - TOPOLOGY
+          - REQUIRE_API_VERSION
+          - MONGODB_VERSION
+          - ORCHESTRATION_FILE
+          - MONGODB_BINARIES
+    - command: expansions.update
+      params:
+        file: mo-expansion.yml
+
+  "stop mongo orchestration":
+    - command: shell.exec
+      params:
+        script: |
+          ${PREPARE_SHELL}
+          cd ${MONGO_ORCHESTRATION_HOME}
+          # source the mongo-orchestration virtualenv if it exists
+          if [ -f venv/bin/activate ]; then
+            . venv/bin/activate
+          elif [ -f venv/Scripts/activate ]; then
+            . venv/Scripts/activate
+          fi
+          mongo-orchestration stop || true
```

Comment:
it boils down to common sense. If the script is too trivial it seems fine to embed it without containing in an external script. But the static-check file is one of the two key tasks in this project (for now), so it deserves a script for better management (internally it will invoke another `java-config,sh` file, so it is not that trivial).

Reply:
It seems we should move common data around using evergreen expansions, since this seems to be a best-practice in evergreen. This should remove the java config file from this file - but this isn't preventing their direct inclusion. The scripts are tiny, and both only used from one place, so it seems pointless to violate locality of behaviour by moving them to a file. This is my sense of things, but I expect it is not common.

I think in cases where the common sense is not obvious, the team should just agree on a convention, because consistency is important. I do not know how to tell if something is a "key task" - whatever convention we choose, it should usually be obvious how to apply it.

Reply:
for such devops task, it is easy to encounter some grey area wherein either way is fine and it is hard to pinpoint which rule to follow. Consistency is always a good conccern though.

I agree that java config stuff could be put into expansion (including `PREPARE_SHELL`). It would simplify a lot. I followed existing pattern from java driver but the legacy pattern is not always the best.

Reply:
Since we could not think of a clear rule, the rule we decided on was to go with whatever the PR author picks (unless there is some compelling technical reason).